### PR TITLE
feat(ui5-menu): add delay between opening and closing submenu and sub menu items

### PR DESCRIPTION
**Pull request by hinzzx**
Previously, the sub-menus would **open** and **close** in an instant when **hovering over** or **hovering out** of a menu item. This sometimes caused unintentional closing of sub-menus, resulting in a poor user experience.

To address this, we made the following changes:

- We added a `300ms` delay when opening a sub-menu upon hovering over it.
- Added `400ms` delay when closing a sub-menu upon hovering out.

Fixes: #7078

### CODE REVIEW

File: packages/main/src/Menu.ts [hunk 1]
```diff
@@ -535,6 +539,40 @@ class Menu extends UI5Element {
...
 	}
 
+	startOpenTimeout(item: MenuItem, opener: OpenerStandardListItem, hoverId: string) {
+		// If theres already a timeout, clears it
+		this.clearOpenTimeout(item);
+
+		// Sets the new timeout
+		this.hoverTimeouts[item.id] = window.setTimeout(() => {
+			this._prepareSubMenuDesktopTablet(item, opener, hoverId);
+		}, 300);
```
Comment (NHristov-sap) [inline packages/main/src/Menu.ts:549]: It will be better if we have this in a constant.

File: packages/main/src/Menu.ts [hunk 2]
```diff
@@ -535,6 +539,40 @@ class Menu extends UI5Element {
...
+	}
+
+	startCloseTimeout(item: MenuItem) {
+		// If theres already a timeout, clears it
+		this.clearCloseTimeout(item);
+
+		// Sets the new timeout
+		this.unhoverTimeouts[item.id] = window.setTimeout(() => {
+			this._closeItemSubMenu(item);
+		}, 400);
```
Comment (NHristov-sap) [inline packages/main/src/Menu.ts:559]: Is there a reason to have different timeouts for opening and closing?
If yes, this also should be a constant.

**Comment by NHristov-sap** [comment]
Looks good to me, but please adapt the test too.

**Review by NHristov-sap — APPROVED** [review]

**Review by nnaydenow — CHANGES_REQUESTED** [review]
At all do we need timeout maps since we clear every existing timeout when the new one is going to be created?

**Comment by nnaydenow** [comment]
Looks good to me. Please add some information about the comment test.

**Review by tsanislavgatev — APPROVED** [review]
We fixed the test, so we are good to go.

**Pull request merged**